pkg/flare/envvars.go

```diff
@@ -66,6 +62,9 @@ func getWhitelistedEnvvars() []string {
 // zipEnvvars collects whitelisted envvars that can affect the agent's
 // behaviour while not being handled by viper
 func zipEnvvars(tempDir, hostname string) error {
+	// update whitelisted env variables with those that are config option bindings
+	envvarNameWhitelist = append(envvarNameWhitelist, config.ConfigEnvVars...)
```

Comment:
🤔 That actually makes it more difficult to test as `TestEnvvarWhitelisting`doesn't benefit from this call anymore.

@xvello what do you think is best between:
- reverting to the `init`
- leaving the update in `zipEnvvars` but adding it as well in the test itself [here](https://github.com/DataDog/datadog-agent/blob/b3fe0acb358dfd48b1f9fed978e797a41e13267f/pkg/flare/envvars_test.go#L40)

Reply:
- linked the go spec in an edit of my init comment, we should indeed be safe relying on `init`, sorry for the false positive
- another option would be to get `getWhitelistedEnvvars` to build the list itself as a local variable (we'll only call it once per flare anyway)